### PR TITLE
Generate default error message

### PR DESCRIPTION
**Pull request by PhilipIngham**
Issue: [#3233](https://github.com/junit-team/junit5/issues/3233)

## Overview
I have added a default error message for the [Assertions.fail()](https://github.com/junit-team/junit5/blob/1c294651ab3965a8bfe8909f6efdc56c5a47c4f0/junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java#L106-L118) function as discussed in the issue, and made some grammatical changes in the surrounding areas. As said in the issue, it's a very minor problem, so nothing is really lost if this doesn't get implemented, but the option's there.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

**Comment by marcphilipp** [comment]
@PhilipIngham Thanks for your PR! However, the team first needs to discuss and decide on #3233 before we can, potentially, proceed.

**Comment by PhilipIngham** [comment]
> @PhilipIngham Thanks for your PR! However, the team first needs to discuss and decide on #3233 before we can, potentially, proceed.

Yeah that's no worries :) Just wanted to contribute something since I use JUnit a lot, and now I can say I have (or at least attempted, even if nothing moves forwards past discussions :) )

File: junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java [hunk 1]
```diff
@@ -73,8 +73,8 @@ public static ReportEntry from(String key, String value) {
 	}
 
 	private void add(String key, String value) {
-		Preconditions.notBlank(key, "key must not be null or blank");
-		Preconditions.notBlank(value, "value must not be null or blank");
+		Preconditions.notBlank(key, "Key must not be null or blank");
```
Comment (pzygielo) [inline junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java:76]: This message seems to refer to parameter, and there is no parameter `Key` now.
It is not consistent with line 55 as well.

Similar for line below.

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java [hunk 2]
```diff
@@ -135,6 +135,9 @@ public static <V> V fail() {
 	 * }</pre>
 	 */
 	public static <V> V fail(String message) {
+		if (message.isBlank()) {
```
Comment (pzygielo) [inline junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java:138]: `AssertionUtils.fail` is fine accepting `null`, but here `fail` will fail with NPEx for `null` message.

Similar below.

File: junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java (hunk 1 above)
Comment (PhilipIngham) [reply junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java]: Makes sense, pushed a fix to revert this

**Comment by marcphilipp** [comment]
Closing with rationale in https://github.com/junit-team/junit5/issues/3233#issuecomment-2265311476

**Pull request closed without merging**